lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -1381,8 +1381,11 @@ namespace Js
             // xplat-todo: Need to replace this with platform-agnostic API
             compareResult = CompareStringEx(givenLocale != nullptr ? givenLocale : defaultLocale, compareFlags, aLeft, size1, aRight, size2, NULL, NULL, 0);
 #else
-            // FIXME (doilij): when CompareStringEx is implemented in PlatformAgnostic, call that function here
-            compareResult = 2; // 2 means strings are equal (reasonable default)
+            compareResult = wcsncmp(aLeft, aRight, min(size1, size2));
+            if (compareResult == 0 && size1 != size2)
+            {
+                compareResult = size1 > size2 ? 1 : -1;
+            }
```

Comment:
You have to add 2 to compareResult because CompareStringEx returns 0 on error, 1 for less, 2 for equal, and 3 for greater. The rest of the code below assumes this.

Reply:
kept the impl. separate and added a comment to make the intention clear  -> `(once ICU interface supports this, keep the implementation below for non-icu!)`